pkg/diagnostics/service_monitoring.go

```diff
@@ -175,8 +174,6 @@ func newServiceMetrics() *serviceMetrics {
 			"The latency of service invocation response.",
 			stats.UnitMilliseconds),
 
-		// TODO: use the correct context for each request
-		ctx:     context.Background(),
```

Comment:
Here too I personally think diagnostics should use a context background. Maybe one with a per-operation timeout, but that is nevertheless un-tied from the context of the parent request. So if the parent request is canceled, we still get the metrics.